language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/workspace/BallerinaWorkspaceManager.java

```diff
@@ -138,6 +138,11 @@ public class BallerinaWorkspaceManager implements WorkspaceManager {
      *  lock does not consider the first compilation (before creating the project context).
      */
     private final Map<Path, Lock> projectLockMap;
+    /**
+     * The build options are used when compiling the project for the LS change events. The build options can be
+     * changed based on the flags set in the client.
+     */
+    private BuildOptions buildOptions;
```

Comment:
Just to clarify, the same build options always apply to all the projects managed by this workspace manager?